tests/core/shader.test.js

```diff
@@ -0,0 +1,261 @@
+/* global AFRAME */
+/* global assert, process, setup, suite, test, teardown */
+var entityFactory = require('../helpers').entityFactory;
+var componentName = 'shader';
+
+var VIDEO = 'https://ucarecdn.com/bcece0a8-86ce-460e-856b-40dac4875f15/'; // 'base/tests/assets/test.mp4';
+
+function isEmpty (map) {
+  for (var key in map) {
+    return !map.hasOwnProperty(key);
+  }
+  return true;
+}
+
+suite(componentName, function () {
+  setup(function (done) {
+    this.el = entityFactory();
```

Comment:
i'd wait for it to load.

`el.addEventListener('loaded', function () { done(); });`